packages/bsky/src/api/app/bsky/feed/getAuthorFeed.ts

```diff
@@ -46,10 +46,23 @@ export default function (server: Server, ctx: AppContext) {
         }
       }
 
+      // defaults to posts, reposts, and replies
       let feedItemsQb = feedService
         .selectFeedItemQb()
         .where('originatorDid', '=', did)
 
+      if (filter === 'posts_with_media') {
+        // only posts with media
+        feedItemsQb = feedItemsQb.innerJoin(
```

Comment:
first thought was to do a `whereExists` here but I actually think the query planner does the same thing either way & the join actually gives it a bit more flexibility

long-winded way of saying: looks good 😛

Reply:
oh actually this does need to be a `whereExists` because the inner join is causing duplicate entries for posts with more than one image embedded

Reply:
Yeah I only did this because it was shorter. Do you think an exists would make the intention a little more obvious though? I kinda do now that I think about it.

Reply:
Changed it to the subquery, I do think it's a little more obvious what's happening for future generations. Cool?